azurerm/internal/services/bot/bot_connection_resource.go

```diff
@@ -112,9 +112,37 @@ func resourceArmBotConnectionCreate(d *schema.ResourceData, meta interface{}) er
 		}
 	}
 
+	serviceProviderName := d.Get("service_provider_name").(string)
+	serviceProviderFound := false
+	var availableProviders []string
+
+	serviceProviders, err := client.ListServiceProviders(ctx)
+	if err != nil {
+		return fmt.Errorf("listing Bot Connection service provider: %+v", err)
+	}
+
+	if serviceProviders.Value == nil {
+		return fmt.Errorf("no available service provider %+v", serviceProviders)
+	}
+	for _, provider := range *serviceProviders.Value {
+		if provider.Properties == nil || provider.Properties.ServiceProviderName == nil {
+			continue
+		}
+		name := provider.Properties.ServiceProviderName
+		if serviceProviderName == *name {
+			serviceProviderFound = true
+			break
+		}
+		availableProviders = append(availableProviders, *name)
+	}
+
+	if !serviceProviderFound {
+		return fmt.Errorf("invalid Service Provider Name. available service providers are: %+v", availableProviders)
```

Comment:
this is likely more readable as a CSV:

```suggestion
		return fmt.Errorf("the Service Provider %q was not found. The available service providers are %s", strings.Join(availableProviders, ","))
```